src/components/side_panel/chart/main_chart_panel/main_chart_panel_store.ts

```diff
@@ -0,0 +1,36 @@
+import { Get } from "../../../../store_engine";
+import { SpreadsheetStore } from "../../../../stores";
+import { Command, UID } from "../../../../types";
+import { SidePanelStore } from "../../side_panel/side_panel_store";
+
+export class MainChartPanelStore extends SpreadsheetStore {
+  private sidePanel = this.get(SidePanelStore);
+
+  figureId: UID | null;
+  panel: "configuration" | "design" = "configuration";
+
+  constructor(get: Get, figureId: UID | null) {
+    super(get);
+    this.figureId = figureId;
+  }
+
+  protected handle(cmd: Command): void {
+    switch (cmd.type) {
+      case "DELETE_FIGURE":
+        if (this.sidePanel.componentTag === "ChartPanel") {
+          this.sidePanel.close();
+        }
+        break;
+      case "SELECT_FIGURE":
+        if (cmd.id) {
+          this.figureId = cmd.id;
+        } else if (this.sidePanel.componentTag === "ChartPanel") {
```

Comment:
if this really happens, the typing of SELECT_FIGURE is wrong. SelectFigureCommand.id should always be defined

Reply:
It's wrong indeed :+1: good catch, thanks

Reply:
that line probably belongs to the release of selection plugin actually ...